django/db/backends/base/schema.py

```diff
@@ -798,7 +799,9 @@ def _create_index_name(self, model, column_names, suffix=""):
             )
         # Else generate the name for the index using a different algorithm
         table_name = model._meta.db_table.replace('"', '').replace('.', '_')
-        index_unique_name = '_%x' % abs(hash((table_name, ','.join(column_names))))
```

Comment:
I looked in more detail at this issue now: The backwards incompatibilities are not as bad as thought, cause Django uses `information_schema` to obtain the indices to drop (for better or worse). This also explains why not many people seem to be affected by this bug cause it generally just works. In that sense I think https://code.djangoproject.com/ticket/24390 is something which we may add for 1.8 (no strong feelings since it is basically a none-issue) but certainly does not warrant a 1.7 backport.

As for https://code.djangoproject.com/ticket/24410 -- it should not have `db_index` since MySQL does not support this -- so the issue also only occurs on missuse of the `db_index` feature.

The only real thing remaining is: index names are not deterministic -- whether this is a feature or a bug is for someone else to decide, all in all this does not belong into 1.7 and is imo not a release blocker for 1.8.